app.js

```diff
@@ -48,23 +47,29 @@ if (!!majorVersion && majorVersion < 14) {
       useCreateIndex: true,
       serverSelectionTimeoutMS: 2500
     })
-    .then(() => ensureSystemSettingsInitiated())
+    .then(async (mg) => {
+      await runMigrations(mg.connection.db, mg.connection.getClient());
+      await ensureSystemSettingsInitiated(container);
+    })
     .then(async () => {
       const port = fetchOctoFarmPort();
+
+      // Shit hit the fan
       if (!port || Number.isNaN(parseInt(port))) {
-        throw new Error("The server database-issue mode requires a numeric port input argument");
+        throw new Error("The OctoFarm server requires a numeric port input argument to run");
       }
 
-      const app = await serveOctoFarmNormally(octoFarmServer);
+      const app = await serveOctoFarmNormally(octoFarmServer, container);
       app.listen(port, "0.0.0.0", () => {
         logger.info(`Server started... open it at http://127.0.0.1:${port}`);
       });
+
+      const autoDiscoveryService = container.resolve(DITokens.autoDiscoveryService);
+      return autoDiscoveryService.searchForDevicesOnNetwork();
     })
     .catch(async (err) => {
       logger.error(err.stack);
```

Comment:
This error catcher is not smart enough. Needs checking for database connection, database authentication, settings, startup or migration problems and serve each error accordingly.